### PR TITLE
CLC-6224 - Update bCourses default theme for New UI

### PR DESCRIPTION
**Pull request by nicolaasmatthijs**
New bCourses skin

### CODE REVIEW

**Comment by johncrossman** [comment]
@nicolaasmatthijs `scss-lint` failures need some love.

File: public/canvas/canvas-customization.js [hunk 1]
```diff
@@ -1,76 +1,213 @@
...
 
+  //////////
+  // UTIL //
+  //////////
+
   /**
-   * Adds Alternative Media collapsible information panel to 'Files' section
+   * Perform a CalCentral API Get request
+   *
+   * @param  {String}     url                 The relative URL of the API request that should be made
```
Comment (redconfetti) [inline public/canvas/canvas-customization.js]: :+1:

File: public/canvas/canvas-customization.js [hunk 2]
```diff
@@ -1,76 +1,213 @@
...
+  addEGrades();
+
+  ////////////////
+  // ADD PEOPLE //
+  ////////////////
+
   /**
    * Adds info alert to the 'People' feature
    * on the 'People' page within a course, show additional info to support adding guests
+   * TODO
```
Comment (redconfetti) [inline public/canvas/canvas-customization.js]: @nicolaasmatthijs Todo?

**Comment by redconfetti** [comment]
@nicolaasmatthijs Travis errors:

```
> calcentral@51.0.0 build-production /home/travis/build/ets-berkeley-edu/calcentral
> gulp build --env production
[06:57:48] Using gulpfile ~/build/ets-berkeley-edu/calcentral/gulpfile.js
[06:57:48] Starting 'build'...
[06:57:48] Starting 'build-clean'...
[06:57:48] Finished 'build-clean' after 45 ms
[06:57:48] Starting 'css'...
[06:57:49] Starting 'browserify'...
[06:57:49] Starting 'images'...
[06:57:49] Starting 'fonts'...
events.js:72
        throw er; // Unhandled 'error' event
              ^
Error: src/assets/stylesheets/_canvas_buttons.scss
Error: Undefined variable: "$bc-canvas-off-black".
        on line 85 of src/assets/stylesheets/_canvas_buttons.scss
>>     color: $bc-canvas-off-black;
   -----------^
    at options.error (/home/travis/build/ets-berkeley-edu/calcentral/node_modules/gulp-sass/node_modules/node-sass/lib/index.js:277:32)
npm ERR! calcentral@51.0.0 build-production: `gulp build --env production`
npm ERR! Exit status 8
npm ERR! 
npm ERR! Failed at the calcentral@51.0.0 build-production script.
npm ERR! This is most likely a problem with the calcentral package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     gulp build --env production
npm ERR! You can get their info via:
npm ERR!     npm owner ls calcentral
npm ERR! There is likely additional logging output above.
npm ERR! System Linux 3.13.0-40-generic
npm ERR! command "/home/travis/.nvm/v0.10.36/bin/node" "/home/travis/.nvm/v0.10.36/bin/npm" "run" "build-production"
npm ERR! cwd /home/travis/build/ets-berkeley-edu/calcentral
npm ERR! node -v v0.10.36
npm ERR! npm -v 1.4.28
npm ERR! code ELIFECYCLE
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /home/travis/build/ets-berkeley-edu/calcentral/npm-debug.log
npm ERR! not ok code 0
Fetching: rainbow-2.1.0.gem (100%)
Successfully installed rainbow-2.1.0
Fetching: sass-3.4.22.gem (100%)
Successfully installed sass-3.4.22
Fetching: scss_lint-0.38.0.gem (100%)
Successfully installed scss_lint-0.38.0
3 gems installed
DEPRECATION WARNING:
Sass 3.5 will no longer support Ruby 1.9.3.
Please upgrade to Ruby 2.0.0 or greater as soon as possible.
src/assets/stylesheets/_canvas_base.scss:67 [W] ColorVariable: Color literals like `#3B7EA1` should only be used in variable declarations; they should be referred to via variable everywhere else.
src/assets/stylesheets/_canvas_base.scss:67 [W] HexNotation: Color `#3B7EA1` should be written as `#3b7ea1`
src/assets/stylesheets/_canvas_base.scss:69 [W] ColorVariable: Color literals like `#2D617C` should only be used in variable declarations; they should be referred to via variable everywhere else.
src/assets/stylesheets/_canvas_base.scss:69 [W] HexNotation: Color `#2D617C` should be written as `#2d617c`
src/assets/stylesheets/_canvas_base.scss:432 [W] BorderZero: `border: 0 is preferred over `border: none`
src/assets/stylesheets/_canvas_buttons.scss:4 [W] HexLength: Color `#333333` should be written as `#333`
src/assets/stylesheets/_canvas_buttons.scss:8 [W] HexLength: Color `#ffffff` should be written as `#fff`
src/assets/stylesheets/_canvas_buttons.scss:30 [W] LeadingZero: `0.2` should be written without a leading zero as `.2`
src/assets/stylesheets/_canvas_course_manage_official_sections.scss:24 [W] ZeroUnit: `0px` should be written without units as `0`
scss-lint failed
```

**Comment by redconfetti** [comment]
New Travis Errors:
```
Running JSHint - detect errors and potential problems in JavaScript code
public/canvas/canvas-customization.js: line 96, col 0, Identifier 'current_user_id' is not in camel case.
public/canvas/canvas-customization.js: line 109, col 0, Identifier 'current_user_id' is not in camel case.
public/canvas/canvas-customization.js: line 118, col 0, Identifier 'current_user_id' is not in camel case.
public/canvas/canvas-customization.js: line 144, col 0, Identifier 'current_user_id' is not in camel case.
public/canvas/canvas-customization.js: line 173, col 0, Identifier 'context_id' is not in camel case.
public/canvas/canvas-customization.js: line 175, col 0, Identifier 'context_id' is not in camel case.
public/canvas/canvas-customization.js: line 217, col 0, Identifier 'add_users' is not in camel case.
public/canvas/canvas-customization.js: line 223, col 0, Identifier 'clickable_element' is not in camel case.
public/canvas/canvas-customization.js: line 226, col 0, Identifier 'clickable_element' is not in camel case.
public/canvas/canvas-customization.js: line 340, col 0, Identifier 'manage_files' is not in camel case.
public/canvas/canvas-customization.js: line 594, col 24, 'response' is already defined.
public/canvas/canvas-customization.js: line 228, col 17, 'externalToolsUrl' is not defined.
public/canvas/canvas-customization.js: line 230, col 15, 'externalToolId' is not defined.
public/canvas/canvas-customization.js: line 320, col 28, '$userErrors' is not defined.
public/canvas/canvas-customization.js: line 223, col 11, 'applyInfoAlert' is defined but never used.

public/canvas/index.js: line 8, col 1, '$' is not defined.
public/canvas/index.js: line 11, col 11, '$' is not defined.
public/canvas/index.js: line 16, col 1, '$' is not defined.

18 errors
rake aborted!
JSHint failed!
```

**Comment by nicolaasmatthijs** [comment]
Pushed fixes for all outstanding issues and JSHint and SCSS errors. There's one final fix I need to push before this is ready for more review.

**Comment by nicolaasmatthijs** [comment]
Implemented remaining improvements and refactoring of "Add People" customizations. This is now ready for more review.

**Comment by raydavis** [comment]
Thank you for all the comments, @nicolaasmatthijs , and for fitting this tough job into your other solo responsibilities.

One last request, if you don't mind... 21 commits is a lot of history to wrangle when managing shared branches or trying to figure out when/why something changed. Would you mind squashing them down before I merge into the master branch?

**Comment by nicolaasmatthijs** [comment]
@raydavis : Absolutely. Squashed them down into a single commit.

**Comment by raydavis** [comment]
Thank you, sir.

**Pull request merged**